benchmarks/python/python_sync_benchmark.py

```diff
@@ -0,0 +1,358 @@
+
+    for i in range(num_of_concurrent_threads):
+        thread = Thread(
+            target=execute_commands,
+            args=(clients, total_commands, data_size, action_latencies),
+            name=f"Worker-{i}"
+        )
+        threads.append(thread)
+        thread.start()
+    
+    for thread in threads:
+        thread.join()
+
+def latency_results(prefix, latencies):
+    result = {}
+    result[prefix + "_p50_latency"] = calculate_latency(latencies, 50)
+    result[prefix + "_p90_latency"] = calculate_latency(latencies, 90)
+    result[prefix + "_p99_latency"] = calculate_latency(latencies, 9)
+    result[prefix + "_average_latency"] = truncate_decimal(mean(latencies))
+    result[prefix + "_std_dev"] = truncate_decimal(np.std(latencies))
+
+    return result
+
+
+def create_clients(client_count, action):
+    return [action() for _ in range(client_count)]
+
+
+def run_clients(
+    clients,
+    client_name,
+    total_commands,
+    num_of_concurrent_threads,
+    data_size,
+    is_cluster,
+):
+    now = datetime.now(timezone.utc).strftime("%H:%M:%S")
+    print(
+        f"Starting {client_name} data size: {data_size} concurrency:"
+        f"{num_of_concurrent_threads} client count: {len(clients)} {now}"
+    )
+    action_latencies = {
+        ChosenAction.GET_NON_EXISTING: list(),
+        ChosenAction.GET_EXISTING: list(),
+        ChosenAction.SET: list(),
+    }
+    time = create_and_run_concurrent_threads(
+        clients, total_commands, num_of_concurrent_threads, data_size, action_latencies
+    )
+    tps = int(started_tasks_counter / time)
+    get_non_existing_latencies = action_latencies[ChosenAction.GET_NON_EXISTING]
+    get_non_existing_latency_results = latency_results(
+        "get_non_existing", get_non_existing_latencies
+    )
+
+    get_existing_latencies = action_latencies[ChosenAction.GET_EXISTING]
+    get_existing_latency_results = latency_results(
+        "get_existing", get_existing_latencies
+    )
+
+    set_latencies = action_latencies[ChosenAction.SET]
+    set_results = latency_results("set", set_latencies)
+
+    json_res = {
+        **{
+            "client": client_name,
+            "num_of_tasks": num_of_concurrent_threads,
+            "data_size": data_size,
+            "tps": tps,
+            "client_count": len(clients),
+            "is_cluster": is_cluster,
+        },
+        **get_existing_latency_results,
+        **get_non_existing_latency_results,
+        **set_results,
+    }
+
+    bench_json_results.append(json_res)
+
+
+def main(
+    total_commands,
+    num_of_concurrent_threads,
+    data_size,
+    clients_to_run,
+    host,
+    client_count,
+    use_tls,
+    is_cluster,
+):
+    if clients_to_run == "all":
+        client_class = redispy.RedisCluster if is_cluster else redispy.Redis
+        clients = create_clients(
+            client_count,
+            lambda: client_class(
+                host=host, port=port, decode_responses=True, ssl=use_tls
```

Comment:
since we return bytes in glide - we shall remove "decode_responses=True", that configures redispy to return strings and not bytes (and thus impact performance)